### ISSUE TITLE
Wrong comparison result in version 1.0.2

### ISSUE THREAD

**Issue by zhangwj618** [Nuitka/Nuitka#1729]
Here is my example script **test.py**:

```
for s in ['1', '123']:
    print(s, len(s), 2 <= len(s) <= 4)
```

Here is the output from the compiled executable:

```shell
$ python -m nuitka test.py
$ ./test.bin
1 1 True
123 3 False
```

I installed version 1.0.2 from pip. I also tried version 1.0.1 and it didn't have this bug.

**Comment by kayhayen** [comment]
Hello,

yes, this is the result of a bug fix I made, 1.0.3 will be coming soon. Sorry for the issues but two bugs were hiding each other, and fixing one while it seemed prudent, may have made the issue worse.

**Comment by kayhayen** [comment]
Fixed in 1.0.3, which I just pushed, PyPI will pick it up during the day.
